src/app/root-store/server-data/vm-snapshots/vm-snapshots.effects.ts

```diff
@@ -211,8 +254,59 @@ export class VmSnapshotsEffects {
   );
 
   @Effect()
-  revertConfirmation$: Observable<Action> = this.actions$.pipe(
+  checkRevertPossibility$: Observable<Action> = this.actions$.pipe(
     ofType<Revert>(VmSnapshotActionTypes.Revert),
+    withLatestFrom(
+      this.store.pipe(select(vmSnapshotSelectors.selectEntities)),
+      this.store.pipe(select(fromVM.selectEntities)),
+    ),
+    map(([action, vmSnapshotsEntities, vmEntities]) => {
+      const vmSnapshot = vmSnapshotsEntities[action.payload.id];
+      const vm = vmEntities[vmSnapshot.virtualmachineid];
+      return {
+        vm,
+        vmSnapshotId: vmSnapshot.id,
+      };
+    }),
+    exhaustMap(({ vmSnapshotId, vm }) => {
+      const params = {
+        resourceid: vmSnapshotId,
+        resourcetype: vmSnapshotEntityName,
+        key: vmSnapshotOfferingTagKey,
+      };
+      return this.tagService.getList(params).pipe(
+        /**
+         * if we unable to verify is revert allowed then we dispatch RevertAllowed in this case
```

Comment:
* If we ARE unable to verify IF revert IS allowed, then we dispatch RevertAllowed
         * In this case, the user will get a server error if the revert is not allowed